app/frontend/actions/send.ts

```diff
@@ -36,16 +38,16 @@ export default (store: Store) => {
     if (state.sendAmount.assetFamily === AssetFamily.TOKEN) {
       const {policyId, assetName, quantity} = state.sendAmount.token
       // TODO: we should have a tokenProvider to get token O(1)
-      const tokenBalance = getSourceAccountInfo(state).tokenBalance.find(
-        (token) => token.policyId === policyId && token.assetName === assetName
-      ).quantity
+      const tokenBalance = getSourceAccountInfo(state)?.tokenBalance?.find(
```

Comment:
`getSourceAccountInfo()` should throw if there's no source account, the original code would fail anyway if the source account isn't defined, so I'd add the assertion right in the declaration of `getSourceAccountInfo()`